src/test/java/de/dhbw/karlsruhe/students/mailflow/unit/ChangeSignatureTest.java

```diff
@@ -0,0 +1,68 @@
+package de.dhbw.karlsruhe.students.mailflow.unit;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import de.dhbw.karlsruhe.students.mailflow.core.application.auth.AuthSessionUseCase;
+import de.dhbw.karlsruhe.students.mailflow.core.application.usersettings.changesignature.ChangeSignatureService;
+import de.dhbw.karlsruhe.students.mailflow.core.application.usersettings.changesignature.LoadSettingsException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Address;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.User;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.UserSettings;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.UserSettingsRepository;
+import de.dhbw.karlsruhe.students.mailflow.external.infrastructure.authorization.SaveSettingsException;
+import org.junit.jupiter.api.Test;
+
+class ChangeSignatureTest {
+
+  UserSettingsRepository mockUserSettingsRepository =
+      new UserSettingsRepository() {
+        @Override
+        public void updateUserSettings(UserSettings userSettings) {}
+
+        @Override
+        public void removeUserSettings(Address address) {}
+
+        @Override
+        public UserSettings getSettings(Address address) {
+          Address testAddress = new Address("test", "example.com");
+          return new UserSettings(testAddress, "Test Signature");
+        }
+      };
+
+  AuthSessionUseCase mockAuthSession =
+      new AuthSessionUseCase() {
+        @Override
+        public boolean isLoggedIn() {
+          return false;
```

Comment:
you mean in line 34. set it to true

Reply:
No you have to check in your ChangeSignatureService that the provided user address is not null. If it is null you should throw an exception